tasks/main.yml

```diff
@@ -2,13 +2,17 @@
 - name: enable ppa
   apt_repository:
     repo: 'ppa:certbot/certbot'
-  when: ansible_pkg_mgr == "apt"
+  when: 
+    - ansible_pkg_mgr == "apt"
+    - ansible_lsb.major_release|int < 20
 
 - name: update apt cache
   apt:
     update_cache: yes
     cache_valid_time: 3600
-  when: ansible_pkg_mgr == "apt"
+  when:
+   - ansible_pkg_mgr == "apt"
+   - ansible_facts.packages["certbot"] is undefined
```

Comment:
couldn't you use the same check for the ppa step?

Reply:
why should we? repo should only be added in ubuntu version 19 and lower.
repo update should only happen if certbot is not already installed, they are two different requirements